umh-core/pkg/service/redpanda_monitor/redpanda_monitor.go

```diff
@@ -0,0 +1,618 @@
+		clusterConfigDataBytes = append(clusterConfigDataBytes, log.Content...)
+	}
+
+	// Remove the START_MARKER, MID_MARKER and END_MARKER
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(START_MARKER), []byte{})
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(MID_MARKER), []byte{})
+	metricsDataBytes = bytes.ReplaceAll(metricsDataBytes, []byte(END_MARKER), []byte{})
+
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(START_MARKER), []byte{})
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(MID_MARKER), []byte{})
+	clusterConfigDataBytes = bytes.ReplaceAll(clusterConfigDataBytes, []byte(END_MARKER), []byte{})
+
+	var wg sync.WaitGroup
+	var metrics *RedpandaMetrics
+	var clusterConfig *ClusterConfig
+	var metricsErr, clusterConfigErr error
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		metrics, metricsErr = s.processMetricsDataBytes(metricsDataBytes, tick)
+	}()
+
+	go func() {
+		defer wg.Done()
+		clusterConfig, clusterConfigErr = s.processClusterConfigDataBytes(clusterConfigDataBytes, tick)
+	}()
+
+	wg.Wait()
+
+	if metricsErr != nil {
+		return nil, fmt.Errorf("failed to process metrics data: %w", metricsErr)
+	}
+
+	if clusterConfigErr != nil {
+		return nil, fmt.Errorf("failed to process cluster config data: %w", clusterConfigErr)
+	}
+
+	return &RedpandaMetricsAndClusterConfig{
+		Metrics:       metrics,
+		ClusterConfig: clusterConfig,
+	}, nil
+}
+
+func (s *RedpandaMonitorService) processMetricsDataBytes(metricsDataBytes []byte, tick uint64) (*RedpandaMetrics, error) {
+
+	// If the data contains "curl: (7)", we can directly abort (connection refused)
+	if strings.Contains(string(metricsDataBytes), "curl: (7)") {
+		return nil, fmt.Errorf("curl error: %s", string(metricsDataBytes))
+	}
+
+	// If we have an error 28, we can directly abort (timeout)
+	if strings.Contains(string(metricsDataBytes), "curl: (28)") {
+		return nil, fmt.Errorf("curl error: %s", string(metricsDataBytes))
+	}
+
+	// Decode the hex encoded metrics data
+	decodedMetricsDataBytes, err := hex.DecodeString(string(metricsDataBytes))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode metrics data: %w", err)
+	}
+
+	// Decompress the metrics data
+	gzipReader, err := gzip.NewReader(bytes.NewReader(decodedMetricsDataBytes))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decompress metrics data: %w", err)
+	}
+	defer gzipReader.Close()
+
+	// Parse the metrics
+	metrics, err := parseMetrics(gzipReader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse metrics: %w", err)
+	}
+
+	// Update the metrics state
+	s.metricsState.UpdateFromMetrics(metrics, tick)
+
+	return &RedpandaMetrics{
+		Metrics:      metrics,
+		MetricsState: s.metricsState,
+	}, nil
+}
+
+func (s *RedpandaMonitorService) processClusterConfigDataBytes(clusterConfigDataBytes []byte, tick uint64) (*ClusterConfig, error) {
+
+	if strings.Contains(string(clusterConfigDataBytes), "curl: (7)") {
```

Comment:
maybe put this into a seperate helper function a this is the same here in both functions